thorlcr/activities/rollup/throllupslave.cpp

```diff
@@ -601,26 +613,35 @@ class CRollupGroupSlaveActivity : public CSlaveActivity, public CThorDataLink
         if (eoi)
             return NULL;
 
-        loop
+        try
         {
-            CThorExpandingRowArray rows(*this, queryRowInterfaces(input));
-            Owned<IRowStream> rowStream = groupLoader->loadGroup(input, abortSoon, &rows);
-            unsigned count = rows.ordinality();
-            if (0 == count)
+            loop
             {
-                eoi = true;
-                return NULL;
-            }
+                Owned<IRowStream> rowStream = groupLoader->loadGroup(input, abortSoon, &rows);
```

Comment:
It will never return a stream in this context.
It must fit into the 'rhs' passed in, or it will fail with OOM trying.
It would only return a stream if it spilt (which it is not allowed to do).

So is probably clearer without the Owned<IRowStream> rowStream, I'll remove.
Similar in a couple of places too, I will also change those.